internal/engine/upper_test.go

```diff
@@ -198,9 +198,18 @@ type fakeBuildAndDeployer struct {
 	nextDeployedUID           types.UID
 	nextPodTemplateSpecHashes []k8s.PodTemplateSpecHash
 
-	// Set this to simulate the build failing. Do not set this directly, use fixture.SetNextBuildFailure
+	// Set this to simulate the build failing.
+	// Do not set this directly, use fixture.SetNextBuildFailure
 	nextBuildFailure error
 
+	// Set this to simulate a live-update user build failure.
+	//
+	// This is slightly different than a normal build failure,
```

Comment:
...wait, doesn't this belie [the comment from above](https://github.com/tilt-dev/tilt/pull/3473/files#diff-febc1ffa49840bcca6121206c409ab75R231) of "if this build failed, [lastResult] will be set to nil"? How do these interact? if there any way to tell just from the BuildState whether the last LiveUpdate succeeded or failed? (I suppose the follow-up question being, is there anywhere we NEED to know that?)

Reply:
did you read the comment in the PR description?

I would honestly prefer we get rid of the whole nomenclature around "success" and "failure" in our build system. Different BuildAndDeployers have different opinions about what they mean in practice. But I think removing that nomenclature is a much bigger and more complex change.

Reply:
(Bazel ran into this problem hard - they split it into "SUCCESS" and "a whole bunch of different non-success statuses")
https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/actions/SpawnResult.java#L36

Reply:
(i particularly :heart: EXECUTION_FAILED vs EXECUTION_FAILED_CATASTROPHICALLY)

Reply:
hmm rereading the PR comment, I think the "if a build failed... set to nil" comment needs to be much more explicit about the new definition of "failed build" / that it only applies to image build results.